src/rdkafka_request.h

```diff
@@ -107,6 +107,41 @@ typedef struct rd_kafkap_NodeEndpoints_s {
         rd_kafkap_NodeEndpoint_t *NodeEndpoints;
 } rd_kafkap_NodeEndpoints_t;
 
+typedef struct rd_kafkap_produce_reply_tags_Partition_s {
+        int32_t Partition;
+        rd_kafkap_CurrentLeader_t CurrentLeader;
+} rd_kafkap_produce_reply_tags_Partition_t;
+
+typedef struct rd_kafkap_produce_reply_tags_Topic_s {
+        char *TopicName;
+        int32_t PartitionCnt;
+        rd_kafkap_produce_reply_tags_Partition_t *PartitionTags;
+} rd_kafkap_produce_reply_tags_Topic_t;
+
+typedef struct rd_kafkap_produce_reply_tags_s {
+        rd_kafkap_NodeEndpoints_t NodeEndpoints;
+        int32_t TopicCnt;
+        rd_kafkap_produce_reply_tags_Topic_t *TopicTags;
+} rd_kafkap_produce_reply_tags_t;
+
+typedef struct rd_kafkap_fetch_reply_PartitionTags_s {
+        int32_t PartitionId;
+        rd_kafkap_CurrentLeader_t CurrentLeader;
+} rd_kafkap_fetch_reply_PartitionTags_t;
+
+typedef struct rd_kafkap_fetch_reply_TopicTags_s {
+        int32_t PartitionCnt;
+        char *TopicName;
+        rd_kafka_Uuid_t TopicId;
+        rd_kafkap_fetch_reply_PartitionTags_t *PartitionTags;
+} rd_kafkap_fetch_reply_TopicTags_t;
```

Comment:
The update op must be sent only for partitions with a leader change, you can keep track with this counter
```suggestion
        int32_t partitions_with_leader_change_cnt;
} rd_kafkap_fetch_reply_TopicTags_t;
```

Reply:
Updated to PartitionsWithLeaderChangedCnt;

Reply:
same

Reply:
Updated